src/org/omegat/gui/exttrans/MachineTranslateTextArea.java

```diff
@@ -147,16 +205,58 @@ private void startSearchThread(SourceTextEntry newEntry, boolean force) {
     @Override
     protected void setFoundResult(final SourceTextEntry se, final MachineTranslationInfo data) {
         UIThreadsUtil.mustBeSwingThread();
-
         if (data != null && data.result != null) {
-            displayed.add (data);
-            setText(getText() + data.result + "\n<" + data.translatorName + ">\n\n");
+            int index = displayed.size();
+            StringBuilder sb = new StringBuilder();
+            sb.append("<div id=\"").append(index).append("\">");
+            sb.append(data.result);
+            sb.append("<div class=\"engine\">&lt;");
+            sb.append(data.translatorName);
+            sb.append("&gt;</div></div>");
+            try {
+                insertElement(data, sb.toString());
+            } catch (BadLocationException | IOException e) {
+                Log.log(e);
```

Comment:
When exception happened, field `displayed` become inconsistent with actual html. Should update `displayed` when all are well.

Reply:
https://github.com/omegat-org/omegat/pull/205/commits/8ba52eda43a2310d13bd29440214c52c6b8dfe92  fix it.